### ISSUE TITLE
Backport testrpc-sc 6.4.5 to solidity-coverage v5

### ISSUE THREAD

**Issue by yxliang01** [sc-forks/solidity-coverage#331]
This issue is a feature request for porting over https://github.com/trufflesuite/ganache-core/pull/13 . Many of my test cases reply on this to run properly, especially when setting a timestamp that is in the past. This functionality can't be achieved without modifying system time back to past for the current version of testrpc-sc.

Thanks

**Comment by cgewecke** [comment]
@yxliang01 Can you try installing `solidity-coverage@beta`? It uses a newer ganache version. If you're launching testrpc-sc separately you'll likely need to add the `--allowUnlimitedContractSize` flag to the command.

**Comment by yxliang01** [comment]
@cgewecke Thanks for the hint. However, our source code is supposed to be tested under solc 0.4 .. So, the `beta` version is not suitable. When checking the https://github.com/sc-forks/testrpc-sc repo, I don't see the branch for testrpc-sc `6.4.1`, would you like to point out where the corresponding source code for `6.4.1` is (the testrpc-sc version `beta` release is using)?

**Comment by cgewecke** [comment]
@yxliang01 Yes you should be able to get the 6.4.1 client separately by running
```
npm install --save-dev ethereumjs-testrpc-sc@beta
```

**Comment by yxliang01** [comment]
@cgewecke We are using source code version of installation due to some customization need. Would you like to point out where the `beta` release source code for `ethereumjs-testrpc-sc` is available?

**Comment by cgewecke** [comment]
@yxliang01 It looks like the latest version is the coverage branch of `ganache-core-sc`, which is at v2.5.3. That consumes a coverage enabled vm - `ethereumjs-vm-sc` whose latest state is also on the coverage branch.

And it's consuming a specific commit of `provider-engine-sc` although I don't think that has any coverage logic in it so may not be necessary, but can't hurt either. 

TLDR to publish your own client you need to fork those three, make your changes, then fork trufflesuite/ganache-cli and make a branch that consumes your forks.

**Comment by yxliang01** [comment]
@cgewecke Thanks a lot for the detailed explanation. It helps. Meanwhile, I would like to confirm that so the source code for `ethereumjs-testrpc-sc@beta`(a.k.a fork of `trufflesuite/ganache-cli`) is not published yet, is it?

**Comment by cgewecke** [comment]
@yxliang01 The source code *is* all the things in the previous comment published to npm as a webpacked bundle. There's nothing else.

**Comment by cgewecke** [comment]
@yxliang01 Do you still want v0.5.x to be published with an updated testrpc or are you going to do this in your own fork?

**Comment by yxliang01** [comment]
@cgewecke Is there any breaking change with this updated testrpc? If not, I don't see why it shouldn't be published with v0.5.x of `solidity-coverage` . If you are willing, backporting would be great! But, in any case, since you have already published the source code of the new testrpc-sc, backporting would be fairly easy for me in my own fork. Thanks.

**Comment by cgewecke** [comment]
@yxliang01 Yes, I think they're quite different. There was a chain fork between 6.1 and 6.4 for example. 

There's also a change in the direct invocation command from:
```
./node_modules/ethereumjs-testrpc-sc/build/cli.node.js # Old
./node_modules/ethereumjs-testrpc-sc/cli.js # New
```

I'm going to think about it a bit....

**Comment by cgewecke** [comment]
@yxliang01 Just tested `0.6.1` against Bancor's contracts which are pinned to solidity`0.4.24` and everything was fine. I *think* you can just upgrade your entire fork to 0.6.0 and everything should work...

**Comment by yxliang01** [comment]
@cgewecke Great! I will schedule it and let you know how it goes :)

**Comment by cgewecke** [comment]
Also tested against AragonOS and it's fine. Closing.
